libs/utils/trace_analysis.py

```diff
@@ -1385,3 +1385,135 @@ def getClusterIdleStateResidency(self, cluster):
         """
         return self._getIdleStateResidency(cluster)
 
+    def _plotIdleStateResidency(self, residencies, entity_name, xmax):
+        """
+        Generate Idle state residency plots for the given entities.
+
+        :param residencies: list of residencies to be plot
+        :type residencies: list(namedtuple(ResidencyData)) - tuple containing:
+            1) as first element, a label to be used as subplot title
+            2) as second element, a dataframe with residency for each idle
+               state
+
+        :param entity_name: name of the entity ('cpu' or 'cluster') used in the
+            figure name
+        :type entity_name: str
+
+        :param xmax: upper bound of x-axes
+        :type xmax: double
+        """
+        n_plots = len(residencies)
+        gs = gridspec.GridSpec(n_plots, 1)
+        fig = plt.figure()
+
+        for idx, data in enumerate(residencies):
+            r = data.residency
+            if r is None:
+                plt.close(fig)
+                return
+
+            axes = fig.add_subplot(gs[idx])
+            is_first = idx == 0
+            is_last = idx+1 == n_plots
+            yrange = 0.4 * max(6, len(r)) * n_plots
+            r.plot.barh(ax = axes, color='g',
+                        legend=False, figsize=(16,yrange))
+
+            axes.set_xlim(0, 1.05*xmax)
+            axes.set_ylabel('Idle State')
+            axes.set_title(data.label)
+            axes.grid(True)
+            if is_last:
+                axes.set_xlabel('Time [s]')
+            else:
+                axes.set_xticklabels([])
+
+            if is_first:
+                legend_y = axes.get_ylim()[1]
+                axes.annotate('Idle State Residency Time', xy=(0, legend_y),
+                              xytext=(-50, 45), textcoords='offset points',
+                              fontsize=18)
+
+        figname = '{}/{}{}_idle_state_residency.png'\
+                  .format(self.plotsdir, self.prefix, entity_name)
+
+        pl.savefig(figname, bbox_inches='tight')
+
+
+    def plotCPUIdleStateResidency(self, cpus=None):
```

Comment:
These plots are quite similar to the plotCPUFrequencyResidency, however in this case we do not support the "relative" representation (i.e. using percentages)... could it make sense to add this mode as well?